lisp/ess-utils.el

```diff
@@ -491,6 +492,11 @@ etc.")
     ["no" (lambda () (interactive) (setq ess-eval-visibly nil))
      :style radio :enable t :selected (eq ess-eval-visibly nil) ]))
 
+(defun ess-derived-mode-p ()
```

Comment:
This will go away once ess-julia-mode derives properly right? If so, should be a private `ess--` function.

Reply:
There's two ways to go about this:

1. (what we're doing now) let ess-julia-mode derive from julia-mode. This has the benefit of using their syntax table, etc but we have to override the keymap with our own
2. derive ess-julia-mode from ess-mode. This basically flips the first situation and now we're using our keymaps by default but would need to override "our" syntax table with theirs.

I hadn't thought about number 2 until you made this comment, but it probably makes sense to do that. In that case, yes this function will eventually go away.

Reply:
Perhaps a crazy thought, but should ESS be a minor mode?

Reply:
I don't think it's totally crazy but am certain it's outside the scope of this PR :-)

Reply:
Agreed, we can keep thinking about it.